rmgweb/database/views.py

```diff
@@ -427,114 +446,544 @@ def solvationEntry(request, section, subsection, index):
+#                    'ref_dict': ref_dict,
+#                    'word_list': word_list})
+
+
+def get_solvation_data_ML(solvent_solute_smiles, calc_dGsolv, calc_dHsolv, calc_dSsolv,
+                          calc_logK, calc_logP, energy_unit):
+    """
+    Returns a dictionary with solvation property data for a given list of solvent-solute SMILES.
+    """
+
+    solvent_solute_smiles_list = solvent_solute_smiles.split()
+
+    dGsolv_required = any(
+        [calc_dGsolv, calc_dSsolv, calc_logK, calc_logP])  # whether or not dGsolv calculation is needed
+    dHsolv_requied = any([calc_dHsolv, calc_dSsolv])  # whether or not dHsolv calculation is needed
+
+    solvation_data_results = {'Input': [],
+                              'solvent SMILES': [],
+                              'solute SMILES': [],
+                              'Error': [],
+                              f'dGsolv298({energy_unit})': [],
+                              f'dHsolv298({energy_unit})': [],
+                              f'dSsolv298({energy_unit}/K)': [],
+                              'logK': [],
+                              'logP': [],
+                              f'dGsolv298 epi.unc.({energy_unit})': [],
+                              f'dHsolv298 epi.unc.({energy_unit})': [],
+                              }
+
+    # get predictions for each given solvent_solute SMILES
+    for solvent_solute in solvent_solute_smiles_list:
+        # initialization
+        solvent_smiles = '-'
+        solute_smiles = '-'
+        Error = '-'
+        dGsolv298 = '-'
+        dGsolv298_epi_unc = '-'
+        dHsolv298 = '-'
+        dHsolv298_epi_unc = '-'
+        dSsolv298 = '-'
+        logK = '-'
+        logP = '-'
+
+        pair_list = solvent_solute.split('_')
+        if not len(pair_list) == 2:
+            Error = 'Unable to process the input'
+        else:
+            solvent_smiles = pair_list[0]
+            solute_smiles = pair_list[1]
+            pair_smiles = [[solvent_smiles, solute_smiles]]
+            if dGsolv_required:
+                try:
+                    avg_pre, epi_unc, valid_indices = dGsolv_estimator(pair_smiles)  # default is in kcal/mol
+                    dGsolv298 = avg_pre[0]
+                    dGsolv298_epi_unc = epi_unc[0]
+                except:
+                    Error = 'Unable to parse the SMILES'
+            if dHsolv_requied:
+                try:
+                    avg_pre, epi_unc, valid_indices = dHsolv_estimator(pair_smiles)  # default is in kcal/mol
+                    dHsolv298 = avg_pre[0]
+                    dHsolv298_epi_unc = epi_unc[0]
+                except:
+                    Error = 'Unable to parse the SMILES'
+            if calc_dSsolv and dGsolv298 != '-' and dHsolv298 != '-':
+                dSsolv298 = (dHsolv298 - dGsolv298) / 298  # default is in kcal/mol/K
+            if calc_logK and dGsolv298 != '-':
+                logK = -dGsolv298 * 4184 / (
```

Comment:
I chose not to use the user's unit preference setting because I think many users will prefer using these tools without having to create an account. Maybe I can also use the user's unit preference setting in the future, but I won't do it for this PR.
Instead, I created a function `convert_energy_unit` that does the unit conversion for the energy in this commit f82f20f.